Fixes to Scouts Loads Summary Measure

mdahlhausen opened the pull request:
Pull request overview
---------------------
 - Fixes #11
- adds a debug mode to the measure with extra reporting to scout_loads_summary
- adds 'floating' as a state point to heating and cooling in os_lib_scout_meters.rb
- many changes to os_lib_heat_transfer.rb:
   - add hot water equipment to internal convective gain and internal radiant gain
   - remove other internal gain
   - add refrigeration gain
   - include air transfer
   - include reference for radiant delay factors
   - break gains into instantaneous convective and radiant, for each load
   - add debug comparison to internal radiant gains calculation in EnergyPlus
   - add details on the solar radiation variables in comments
   - fixed issue where internal surface loads were getting re-attributed back to those same surfaces, and instead allocate them to exterior surfaces in the space. The new method functionally assumes no inter zone conduction.
   - discounts delayed solar gains, delayed internal gains, and internal surface convection from exterior surfaces

Further changes by @eringold:
  - Implement measure as python EMS
  - postprocessing changes to incorporate new variables

### Pull Request Author
This pull request makes changes to (select all the apply):
 - [x] Reporting Measures

Author pull request checklist:
 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Added tests for new measures
 - [x] Updated measure .xml(s)
 - [x] Register values added to `comstock_column_definitions.csv`
 - [x] 10k+ test run
 - [x] Measure documentation written
 - [x] Changes reflected in example `.yml` files
 - [x] Changes reflected in `README.md` files
 - [x] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`
 - [NA] All new and existing tests pass the CI

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

mdahlhausen committed:
fixes to the load summary measure

mdahlhausen committed:
Merge branch 'main' into fix/scout_loads_summary

mdahlhausen committed:
switch to space lookup for enduse meters

eringold commented on measures/scout_loads_summary/resources/os_lib_scout_meters.rb:
I think you need to update the DistrictHeating meter names to [DistrictHeatingWater](https://bigladdersoftware.com/epx/docs/23-2/input-output-reference/group-simulation-parameters.html#metercustom)

eringold committed:
remove tests from ci

eringold commented:
@mdahlhausen I removed the tests from c.i. because 1) this isn't a workflow measure, 2) the tests are useful for debugging the measure but aren't doing any automated checks except for run completion, and 3) they take a long time and generate big output files. 
Also I still see a number of failures, looks like mostly from simulation failures from bad meter specification:
```
Finished in 1743.899579s, 0.0115 runs/s, 1.0557 assertions/s.

  1) Failure:
ScoutLoadsSummary_Test#test_restaurant [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  2) Failure:
ScoutLoadsSummary_Test#test_district_cooling [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  3) Failure:
ScoutLoadsSummary_Test#test_scn [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  4) Failure:
ScoutLoadsSummary_Test#test_large_office_vav [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  5) Failure:
ScoutLoadsSummary_Test#test_large_office [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  6) Failure:
ScoutLoadsSummary_Test#test_small_office_high_loads [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  7) Failure:
ScoutLoadsSummary_Test#test_medium_office [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  8) Failure:
ScoutLoadsSummary_Test#test_doas_with_vrf [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

  9) Failure:
ScoutLoadsSummary_Test#test_research_special [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

 10) Failure:
ScoutLoadsSummary_Test#test_small_office [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

 11) Failure:
ScoutLoadsSummary_Test#test_district_heating [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

 12) Failure:
ScoutLoadsSummary_Test#test_mfm [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

 13) Failure:
ScoutLoadsSummary_Test#test_baseboard_electric [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

 14) Failure:
ScoutLoadsSummary_Test#test_secondary_school [C:/Repos/NREL/ComStock/measures/scout_loads_summary/tests/Measure_Test.rb:168]:
Expected: "Success"
  Actual: "Fail"

20 runs, 1841 assertions, 14 failures, 0 errors, 0 skips
```
Measure seems to still be a work in progress. Leaving open for now.

mdahlhausen committed:
Merge branch 'main' into fix/scout_loads_summary

mdahlhausen committed:
Merge branch 'fix/scout_loads_summary' of https://github.com/NREL/ComStock into fix/scout_loads_summary

eringold committed:
test model updates

eringold committed:
add debug support to test

eringold committed:
use nonsolar RTS

eringold committed:
window transmitted solar uses space name

eringold committed:
use energy variables

eringold committed:
fix internal mass not being subtracted

eringold committed:
is redistribution happening correctly?

eringold committed:
temp debug helpers

eringold committed:
marginal performance improvements

eringold committed:
test troubleshooting switches

eringold committed:
add internal surface gains to external surfaces, dont subtract

eringold committed:
Merge branch 'main' of https://github.com/NREL/ComStock into fix/scout_loads_summary

eringold committed:
postprocess load components data

mdahlhausen committed:
Energy meter updates

mdahlhausen committed:
add gas equipment

eringold committed:
Merge branch 'fix/scout_loads_summary' of https://github.com/NREL/ComStock into fix/scout_loads_summary

mdahlhausen committed:
add natural gas equipment meters

eringold committed:
add supply gas quipment

eringold committed:
add weighted load component cols to long

eringold committed:
test flags

eringold committed:
avoid double counting by apportioning load based on total heating+cooling energy percentages, not individual

eringold committed:
Merge branch 'fix/scout_loads_summary' of https://github.com/NREL/ComStock into fix/scout_loads_summary

eringold committed:
add comstock measure tests

eringold committed:
enable district heating fuel metering

eringold committed:
minor update to loads reporting

eringold committed:
generalize dview formatting of output

eringold committed:
add missing window IR and shade convective gain, reorganize delayed radiant calcs, reorder calculations

bonnema committed:
add generated_files/, reports/, run/, and out.osw to .gitignore

bonnema committed:
first commit of python plugin load summary; all handles returning -1

bonnema committed:
simple test using openstudio workflow

bonnema committed:
Merge branch 'fix/scout_loads_summary' of github.com:NREL/ComStock into fix/scout_loads_summary

bonnema committed:
use ERB template for python, code is still very much in a testing phase

bonnema committed:
add in.py to .gitignore

bonnema committed:
convert to E+ measure until OS 3.10, 1-model test works, next step calcs

bonnema committed:
assign out vars to py var, add full list of py vars, add py vars by mode

bonnema committed:
add zone name dict, add lists by py var and op mode, sum to make globals

bonnema committed:
add surface names

bonnema committed:
running script with radiant time series calcs added for internal gains

mdahlhausen committed:
adjust ext light measure method calls to new os stds

mdahlhausen committed:
update os stds method call to add space loads

space_type_apply_internal_loads no longer adds infiltration

mdahlhausen committed:
update ext lighting to use new stds method

Add typical exterior lighting.
Right now there is only one default option. Most exterior lighting is parking lighting, and ComStock used the same value across most vintages.
Update once there is variation in standards for exterior lighting based on technology level.

eringold committed:
new measure

eringold committed:
refactored script

eringold committed:
correct system timestep vars and window calcs

eringold committed:
update old measure for validation

eringold committed:
cleanup and runperiod output

eringold committed:
another test file

mdahlhausen committed:
update to OS 3.10.0

mdahlhausen committed:
dockerfile updates for OS 3.10.0

eringold committed:
update script to compute multipliers and include ref load, cleanup measure and add tests

eringold committed:
update xml

eringold committed:
Merge remote-tracking branch 'origin' into fix/scout_loads_summary

eringold committed:
make write dir

eringold committed:
get handles at the right place

eringold committed:
add timeseries output

eringold committed:
Merge branch 'main' of https://github.com/NREL/ComStock into fix/scout_loads_summary

eringold committed:
update postprocessing to weight loads results and output long data

eringold committed:
update loads column defs

mdahlhausen committed:
fix missing district emissions in postproc

- add district emissions columns, add to naming mixin, and list downselect option in log

mdahlhausen committed:
Update comstock_to_cbecs_comparison.py

Don't try to plot end use stacked bar comparisons unless there are at least 2 discrete values in the groupby column. For now, this means 2 different comstock runs. May edit later to accommodate CBECS.

mdahlhausen committed:
Merge branch 'fix/missing_district_postproc' into fix/scout_loads_summary

eringold committed:
add climate zone group to loads long

eringold committed:
remove breakpoint

mdahlhausen committed:
Merge branch 'main' into fix/scout_loads_summary

mdahlhausen committed:
split out refrigeration equipment in loads summary

- split out refrigeration equipment from equipment in the loads_summary measure
- renable tests

mdahlhausen committed:
remove tests without models

remove tests without models saved in the testing directory

mdahlhausen committed:
add copyrights and update .xml

mdahlhausen committed:
add refrigeration equipment loads to postproc

mdahlhausen committed:
add ref equip load intensity

mdahlhausen committed:
load components long csv template

- add missing ref_equip_gain variables to comstock_column_definitions.csv
- create template for exporting the load components in long data format
- update comstock.py output_dir to output_dir["fs_path"] where it was expecting the path and hadn't already been updated

mdahlhausen committed:
update postproc template

mdahlhausen commented:
@eringold @ChristopherCaradonna this measure PR is ready for a final review! Tested and now have a full baseline run with results.

mdahlhausen committed:
add adiabatic comparison test

mdahlhausen committed:
account for net thermal radiation exchange

eringold committed:
Merge pull request #411 from NREL/fix/load_components_radiation

Account for net thermal radiation exchange in load components
